packages/hash/api/src/db/adapter.ts

```diff
@@ -240,11 +240,11 @@ export interface DbClient {
   createEntity(params: {
     accountId: string;
     createdByAccountId: string;
-    entityId?: string | null | undefined;
-    entityVersionId?: string | null | undefined;
+    entityId?: string;
+    entityVersionId?: string;
     entityTypeId?: string;
-    entityTypeVersionId?: string | null | undefined;
-    systemTypeName?: SystemType | null | undefined;
+    entityTypeVersionId?: string;
+    systemTypeName?: SystemType;
```

Comment:
I think the original type signature here was influenced by the fact that the db adapter method was directly being passed GQL params, this is no longer the case so have removed that.

Reply:
I don't think we should pass `undefined` explicitly to any of the parameters, as it doesn't have any more meaning than not providing the parameter at all.

I'm not sure what would happen in a situation like this:
```TypeScript
client.createEntity({
  systemTypeName: systemTypeName ?? undefined,
})
```
I don't think we're doing this anywhere for this method, but I'm sure we're doing it somewhere - and I guess this would break.